### PR TITLE
comments and minor refactor

### PR DESCRIPTION
**Pull request by replay**

### CODE REVIEW

File: slab.go [hunk 1]
```diff
@@ -33,47 +46,55 @@ func newSlab(objSize uint8, objsPerSlab uint) (*slab, error) {
...
-	// get the byte slice header of bitset's data property
+	// get the byte slice header of BitSets data property
 	bitSetDataSlice := (*reflect.SliceHeader)(unsafe.Pointer(&data[1+offsetOfBitSetData]))
 
-	// set the data pointer to point at the address right after the bitSet struct
+	// set the data pointer to point at the address right after the BitSet instance
 	bitSetDataSlice.Data = uintptr(unsafe.Pointer(&data[1+int(sizeOfBitSet)]))
 
-	// return the data byte slice as a slab
+	// return the data byte slice casted as a slab
```
Comment (robert-milan) [inline slab.go]:
```suggestion
	// return the data byte slice converted to a slab pointer
```

File: slab.go [hunk 2]
```diff
@@ -33,47 +46,55 @@ func newSlab(objSize uint8, objsPerSlab uint) (*slab, error) {
...
 	return (*slab)(unsafe.Pointer(&data[0])), nil
 }
 
+// addr returns this slabs address as a SlabAddr type
```
Comment (robert-milan) [inline slab.go]:
```suggestion
// addr returns this slabs' address as a SlabAddr type
```

File: slab.go [hunk 3]
```diff
@@ -33,47 +46,55 @@ func newSlab(objSize uint8, objsPerSlab uint) (*slab, error) {
...
 func (s *slab) addr() SlabAddr {
 	return SlabAddr(unsafe.Pointer(s))
 }
 
+// bitSet returns this slabs BitSet as a pointer to a BitSet
```
Comment (robert-milan) [inline slab.go]:
```suggestion
// bitSet returns this slabs' BitSet as a pointer
```

File: slab.go [hunk 4]
```diff
@@ -95,52 +118,59 @@ func (s *slab) getObjIdx(obj ObjAddr) uint {
 	return uint(objectOffset / uintptr(s.objSize))
 }
 
+// addObj takes an object and adds it to this slice if there is
+// free space for it
+// On success the first return value is the ObjAddr of the newly
+// added object and the second value is true
+// On failure the secpnd return value is false
```
Comment (robert-milan) [inline slab.go]:
```suggestion
// On failure the second return value is false
```

File: slab_pool.go [hunk 5]
```diff
@@ -25,50 +26,60 @@ func NewSlabPool(objSize uint8, objsPerSlab uint) *slabPool {
...
+// slab and then use that one
+// The first return value is the ObjAddr of the added object
+// The second value is the slab address if the call created a new slab
+// If no new slab has been created, then the second value is 0
+// The third value is nil if there was no error, otherwise it is the error
 func (s *slabPool) add(obj []byte) (ObjAddr, SlabAddr, error) {
 	var success bool
 	var objAddr ObjAddr
-	var currentSlab *slab
+	var currentSlab *sla
```
Comment (robert-milan) [inline slab_pool.go]:
```suggestion
	var currentSlab *slab
```

File: slab_pool.go [hunk 6]
```diff
@@ -25,50 +26,60 @@ func NewSlabPool(objSize uint8, objsPerSlab uint) *slabPool {
...
+	// the second return value
 	return objAddr, currentSlab.addr(), nil
 }
 
-// findSlabByObjAddr takes an object address and finds the correct slab
-// where this object is in by looking it up from its slab list
-// it returns the slab index if the correct slab was found, otherwise
-// the return value will be set to the number of known slabs
+// findSlabByObjAddr takes an object address or slab address and then
+// finds the slab where this object is inside by looking it up from
```
Comment (robert-milan) [inline slab_pool.go]:
```suggestion
// finds the slab where this object exists by looking it up from
```

File: slab_pool.go [hunk 7]
```diff
@@ -25,50 +26,60 @@ func NewSlabPool(objSize uint8, objsPerSlab uint) *slabPool {
...
+// its slab list.
+// It returns the slab index if the correct slab was found, otherwise
+// the return value is the number of known slabs.
+// For the lookup to succeed it relies on s.slabs to be sorted
```
Comment (robert-milan) [inline slab_pool.go]:
```suggestion
// For the lookup to succeed it relies on s.slabs to be sorted in descending order
```

File: slab_pool.go [hunk 8]
```diff
@@ -86,12 +98,14 @@ func (s *slabPool) addSlab() (*slab, error) {
 	return addedSlab, nil
 }
 
-// search searches for a byte slice that must have the length of
-// the slab's objectSize.
+// search searches for a byte slice with the length of
+// this slabs objectSize.
```
Comment (robert-milan) [inline slab_pool.go]:
```suggestion
// this slab's objectSize.
```

File: slab_pool.go [hunk 9]
```diff
@@ -115,9 +129,21 @@ func (s *slabPool) search(searching []byte) (ObjAddr, bool) {
...
+// The returned value is a slice of ObjAddr which always has the same length
+// as the slice of searched objects.
+// If a searched object has been found then its address is at the same index
+// in the returned slice as it was in the search slice.
+// If a searched object has not been found, then the value in the returned
+// slice is 0 at the index of the searched object.
 func (s *slabPool) searchBatched(searching [][]byte) []ObjAddr {
 	wg := sync.WaitGroup{}
+
+	// preallocate the resul set that will be returned
```
Comment (robert-milan) [inline slab_pool.go]:
```suggestion
	// preallocate the result set that will be returned
```

File: slab_pool.go (hunk 5 above)
Comment (replay) [reply slab_pool.go]: fixed already :+1:

**Pull request merged**